### PR TITLE
Improve documentation for contributors/maintainers

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

It was pointed out recently that the link to the Contributing guide is buried at the bottom of the README. This commit makes it more prominent by moving it to the top and lists the major sections of the document.

However, this commit also updates the Contributing guide to be more generally helpful and to answer questions that I've received from other teams not familiar with this repo who are unaware that codeownership and releases are shared among teams.

- **[View `README.md`](https://github.com/MetaMask/core/blob/update-contributing-docs/README.md)**
- **[View `contributing.md`](https://github.com/MetaMask/core/blob/update-contributing-docs/docs/contributing.md)**

## References

As mentioned above, this PR was inspired by a message in Slack, but also inspired by conversations I've had in the past with other teams.

## Changelog

N/A

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [x] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,9 +2,21 @@
 
 This monorepo is a collection of packages used across multiple MetaMask clients (e.g. [`metamask-extension`](https://github.com/MetaMask/metamask-extension/), [`metamask-mobile`](https://github.com/MetaMask/metamask-mobile/)).
 
-## Modules
+## Contributing
```
Comment (mcmire) [inline README.md:5]: This section was previously buried at the bottom.

File: docs/contributing.md [hunk 2]
```diff
@@ -148,120 +173,117 @@ To use a preview build for a package within a project, you need to override the
 
 4. Run `yarn install`.
 
-## Adding new packages
+## Releasing changes
```
Comment (mcmire) [inline docs/contributing.md:176]: This section got moved up, and expanded to give people more guidance than before.

File: docs/contributing.md [hunk 3]
```diff
@@ -148,120 +173,117 @@ To use a preview build for a package within a project, you need to override the
...
-5. **"Squash & Merge" the release.**
+   Once the `npm-publishers` team has approved the release, you can click on the link in the Slack message to monitor the remainder of the process.
 
-   This step triggers the [`publish-release` GitHub action](https://github.com/MetaMask/action-publish-release) workflow to tag the final release commit and publish the release on GitHub.
+   Once the action has completed, [check NPM](https://npms.io/search?q=scope%3Ametamask) to verify that all relevant packages has been published.
 
-   Pay attention to the box you see when you press the green button and ensure that the final name of the commit follows the pattern "Release \<new version\>".
+   You're done!
+
+## Performing operations across the monorepo
```
Comment (mcmire) [inline docs/contributing.md:244]: I didn't make any changes to this section, so it should be the same.

File: docs/contributing.md [hunk 4]
```diff
@@ -148,120 +173,117 @@ To use a preview build for a package within a project, you need to override the
...
+> **Note**
+>
+> - `workspaceName` in the Yarn documentation is the `name` field within a package's `package.json`, e.g., `@metamask/address-book-controller`, not the directory where it is located, e.g., `packages/address-book-controller`.
+> - `commandName` in the Yarn documentation is any sub-command that the `yarn` executable would usually take. Pay special attention to the difference between `run` vs `exec`. If you want to run a package script, you would use `run`, e.g., `yarn workspace @metamask/address-book-controller run changelog:validate`; but if you want to run _any_ shell command, you'd use `exec`, e.g. `yarn workspace @metamask/address-book-controller exec cat package.json | jq '.version'`.
 
-   Once the `publish-npm` job has finished, [check NPM](https://npms.io/search?q=scope%3Ametamask) to verify that all relevant packages has been published.
 
-### Handling common errors
```
Comment (mcmire) [inline docs/contributing.md:246]: I'm not sure the "Handling common errors" section is that useful. I've found that people reach out to me if they have issues. We can figure out a way to add it back if we need to.

**Review by mikesposito — APPROVED** [review]
Looks good!

**Pull request merged**